src/porepy/viz/exporter.py

```diff
@@ -520,546 +1178,889 @@ def _export_grid(
-        # cell id map
-        cell_id = {cell_type: np.empty(num_cells, dtype=int)}  # type: ignore
-        cell_pos = 0
+        # Dictionary storing cell->nodes connectivity information
+        cell_to_nodes: Dict[str, np.ndarray] = {cell_type: np.empty((0, 2), dtype=int)}
+
+        # Dictionary collecting all cell ids for each cell type.
+        # Since each cell is a line, the list of cell ids is trivial
+        total_num_cells = np.sum(np.array([g.num_cells for g in gs]))
+        cell_id: Dict[str, List[int]] = {
+            cell_type: np.arange(total_num_cells, dtype=int).tolist()
+        }
 
-        # points
+        # Data structure for storing node coordinates of all 1d grids.
         num_pts = np.sum([g.num_nodes for g in gs])
         meshio_pts = np.empty((num_pts, 3))  # type: ignore
-        pts_pos = 0
 
-        # loop on all the 1d grids
+        # Initialize offset. All data associated to 1d grids is stored in
+        # the same vtu file, essentially using concatenation. To identify
+        # each grid, keep track of number of nodes for each grid.
+        nodes_offset = 0
+
+        # Loop over all 1d grids
         for g in gs:
-            # save the points information
-            sl = slice(pts_pos, pts_pos + g.num_nodes)
+
+            # Store node coordinates
+            sl = slice(nodes_offset, nodes_offset + g.num_nodes)
             meshio_pts[sl, :] = g.nodes.T
 
-            # Cell-node relations
-            g_cell_nodes = g.cell_nodes()
-            g_nodes_cells, g_cells, _ = sps.find(g_cell_nodes)
-            # Ensure ordering of the cells
-            g_nodes_cells = g_nodes_cells[np.argsort(g_cells)]
+            # Lines are 1-simplices, and have a trivial connectvity.
+            cn_indices = self._simplex_cell_to_nodes(1, g)
 
-            # loop on all the grid cells
-            for c in np.arange(g.num_cells):
-                loc = slice(g_cell_nodes.indptr[c], g_cell_nodes.indptr[c + 1])
-                # get the local nodes and save them
-                cell_to_nodes[cell_type][cell_pos, :] = g_nodes_cells[loc] + pts_pos
-                cell_id[cell_type][cell_pos] = cell_pos
-                cell_pos += 1
+            # Add to previous connectivity information
+            cell_to_nodes[cell_type] = np.vstack(
+                (cell_to_nodes[cell_type], cn_indices + nodes_offset)
+            )
 
-            pts_pos += g.num_nodes
+            # Update offsets
+            nodes_offset += g.num_nodes
 
-        # construct the meshio data structure
-        num_block = len(cell_to_nodes)
-        meshio_cells = np.empty(num_block, dtype=object)
-        meshio_cell_id = np.empty(num_block, dtype=object)
+        # Construct the meshio data structure
+        num_blocks = len(cell_to_nodes)
+        meshio_cells = np.empty(num_blocks, dtype=object)
+        meshio_cell_id = np.empty(num_blocks, dtype=object)
```

Comment:
My clear preference in these cases is to not use np.array as general purpose containers, but rather use List. From what I can tell, this should be no problem from the meshio side, or have I missed something?

Reply:
Done, also regarding similar comments.